aws/modules/consolidate-logs/module.ftl

```diff
@@ -0,0 +1,226 @@
+            "Default" : "log"
+        },
+        {
+            "Names" : "deploymentProfile",
+            "Type" : STRING_TYPE,
+            "Description" : "The deployment profile id to enable log consolidation on."
+        },
+        {
+            "Names" : "lambdaSourceUrl",
+            "Type" : STRING_TYPE,
+            "Description" : "A URL to the lambda zip package for sending alerts",
+            "Default" : "https://github.com/hamlet-io/lambda-log-processors/releases/download/v1.0.2/cloudwatch-firehose.zip"
+        },
+        {
+            "Names" : "lambdaSourceHash",
+            "Type" : STRING_TYPE,
+            "Description" : "A sha1 hash of the lambda image to validate the correct one",
+            "Default" : "3a6b1ce462aaa203477044cfe83c66f128381434"
+        },
+        {
+            "Names" : "tier",
+            "Type" : STRING_TYPE,
+            "Description" : "The tier to use to host the components",
+            "Default" : "mgmt"
+        }
+    ]
+/]
+
+[#macro aws_module_consolidatelogs
+    namePrefix
+    deploymentProfile
+    lambdaSourceUrl
+    lambdaSourceHash
+    tier]
+
+    [@debug message="Entering Module: consolidate-logs" context=layerActiveData enabled=false /]
+
+    [#local lambdaName = formatName(namePrefix, "lambda")]
+    [#local datafeedName = formatName(namePrefix, "datafeed")]
+
+    [#local product = getActiveLayer(PRODUCT_LAYER_TYPE) ]
+    [#local environment = getActiveLayer(ENVIRONMENT_LAYER_TYPE)]
+    [#local segment = getActiveLayer(SEGMENT_LAYER_TYPE)]
+
+    [#local namespace = formatName(product["Name"], environment["Name"], segment["Name"])]
+    [#local lambdaSettingNamespace = formatName(namespace, lambdaName)]
+
+    [@loadModule
+        settingSets=[
+           {
+                "Type" : "Settings",
+                "Scope" : "Products",
+                "Namespace" : lambdaSettingNamespace,
+                "Settings" : {
+
+                }
+            } 
+        ]
+        blueprint={
+            "Tiers" : {
+                tier : {
+                    "Components" : {
+                        datafeedName : {
+                            "datafeed": {
+                                "Instances": {
+                                    "default": {
+                                        "DeploymentUnits": [ datafeedName ]
+                                    }
+                                },
+                                "Encrypted": true,
+                                "Destination": {
+                                    "Link": {
+                                        "Tier": "mgmt",
+                                        "Component": "baseline",
+                                        "DataBucket": "opsdata",
+                                        "Instance": "",
+                                        "Version": ""
+                                    }
```

Comment:
IS there any path control as to what top level prefix the logs will end up under?

Reply:
The datafeed component has a configuration option to control the two prefixes required for datafeed 

 Bucket.Prefix - Where the actual data will land 
Bucket.ErrorPrefix - Where failed data will land ( this is usually used with non S3 backends but still needs to be configured ) 

Should we make this Prefix `CWLogs/Logs/`  and ErrorPrefix `CWLogs/Errors/`  ? with https://github.com/hamlet-io/engine-plugin-aws/pull/157/files that will also include the data feed full path after the initial prefix so we can have multiple cwlogdata feeds if required

Reply:
I have added this in https://github.com/hamlet-io/engine-plugin-aws/pull/165/commits/79c8e98b10b059fed45b7b9bb78c9e756229eff4 - we can discuss further once we see how this handles.